Public/Src/Pips/Dll/Operations/Process.cs

```diff
@@ -246,6 +246,30 @@ public sealed partial class Process : Pip
         [PipCaching(FingerprintingRole = FingerprintingRole.Semantic)]
         public RegexDescriptor ErrorRegex { get; }
 
+        /// <summary>
+        /// When false (or not set): process output is scanned for error messages line by line;
+        /// 'errorRegex' is applied to each line and if ANY match is found the ENTIRE line is reported.
+        /// 
+        /// When true: process output is scanned in chunks of up to 1000 lines; 'errorRegex' is applied to
+        /// each chunk and only the matches are reported.Furthermore, if 'errorRegex' contains a capture
+        /// group named "ErrorMessage", the value of that group is reported; otherwise, the value of the
+        /// entire match is reported.
+        /// 
+        ///   NOTE: because this scanning is done against chunks of text (instead of the entire process output),
+        ///         false negatives are possible if an error message spans across multiple chunks.  The scanning
+        ///         is done in chunks because attempting to construct a single string from the entire process
+        ///         output can easily lead to OutOfMemory exception.
+        /// </summary>
+        /// <remarks>
+        /// Regarding fingerprinting: <see cref="ErrorRegex"/> and <see cref="WarningRegex"/> are
+        /// currently part of the process fingerprint, even though those two properties cannot affect
+        /// the outcome (success vs failure) of the process.  This is kind of strange and should probably

```

Comment:
The warning regex is special and should be part of the fingerprint since you could have /warnaserror on. Even without that, the outcome presented to the user could differ based on the warning regex. We evaluate the regex when the pip misses and store some state as to whether a warning should be replayed on a subsequent run so that needs to be in the fingerprint.

It should be safe to remove the error regex from the fingerprint though

Reply:
I'll update the comment